contracts/wrapper/NameWrapper.sol

```diff
@@ -491,11 +499,21 @@ contract NameWrapper is
     ) public {
         bytes32 node = _makeNode(parentNode, labelhash);
         _checkFusesAreSettable(node, fuses);
+        // even if expired, setChildFuses can set to previous owner. If not expired it always sets to current owner.
         (address owner, uint32 oldFuses, uint64 oldExpiry) = getData(
             uint256(node)
         );
+
+        if (oldExpiry < block.timestamp) {
+            oldFuses = 0;
+        }
         // max expiry is set to the expiry of the parent
         (, uint32 parentFuses, uint64 maxExpiry) = getData(uint256(parentNode));
+
+        if (maxExpiry < block.timestamp) {
```

Comment:
I don't think this is necessary, because if `maxExpiry` has already passed, you can't set fuses on the child name anyway.